resolwe/flow/management/commands/list_docker_images.py

```diff
@@ -96,18 +98,20 @@ def handle(self, *args, **options):
 
             # Pull each image
             for img in unique_docker_images:
-                ret = subprocess.call(
+                ret = subprocess.run(
                     shlex.split("{} pull {}".format(docker, img)),
-                    stdout=None if verbosity > 0 else subprocess.DEVNULL,
-                    stderr=None if verbosity > 0 else subprocess.DEVNULL,
+                    stdout=subprocess.PIPE,
+                    stderr=subprocess.STDOUT,
                 )
 
                 # Update set of pulled images.
                 with PULLED_IMAGES_LOCK:
                     PULLED_IMAGES.add(img)
 
-                if ret != 0:
-                    errmsg = "Failed to pull Docker image '{}'!".format(img)
+                if ret.returncode != 0:
+                    errmsg = "Failed to pull Docker image '{}': {}!".format(
+                        img, ret.stdout
```

Comment:
Is stdout a single line? If not, I would prefer to add a new line (or two) before it and remove the exclamation mark.

Reply:
It depends. It could be of arbitrary length.